src/algorithms/tracking/TrackSeeding.cc

```diff
@@ -276,23 +271,34 @@ std::pair<float, float> eicrecon::TrackSeeding::findPCA(std::tuple<float,float,f
   return std::make_pair(xmin,ymin);
 }
 
-int eicrecon::TrackSeeding::determineCharge(std::vector<std::pair<float,float>>& positions) const
+int eicrecon::TrackSeeding::determineCharge(std::vector<std::pair<float,float>>& positions, const std::pair<float,float>& PCA, std::tuple<float,float,float>& RX0Y0) const
 {
-  // determine the charge by the bend angle of the first two hits
+
   int charge = 1;
+
   const auto& firstpos = positions.at(0);
-  const auto& secondpos = positions.at(1);
+  auto hit_x = firstpos.first;
+  auto hit_y = firstpos.second;
+
+  auto xpos = PCA.first;
+  auto ypos = PCA.second;
 
-  const auto firstphi = atan2(firstpos.second, firstpos.first);
-  const auto secondphi = atan2(secondpos.second, secondpos.first);
-  auto dphi = secondphi - firstphi;
-  if(dphi > M_PI) dphi -= 2.*M_PI;
-  if(dphi < -M_PI) dphi += 2*M_PI;
-  if(dphi < 0) charge = -1;
+  float X0 = std::get<1>(RX0Y0);
+  float Y0 = std::get<2>(RX0Y0);
+
+  Acts::Vector2 radial(X0-xpos, Y0-ypos);
+  Acts::Vector2 hit(hit_x-xpos, hit_y-ypos);
+
+  Eigen::Matrix2d  matrix;
+  matrix << radial(0), radial(1), hit(0), hit(1);
+  auto det = matrix.determinant();
```

Comment:
Please use `Vector3::cross` instead of `Matrix2d::determinant`, then introduce `Acts::Vector3 B_hat(0., 0., 1.);` and do a dot product with it.